### PR TITLE
Clear failed images from post draft when posting

### PR DESCRIPTION
**Pull request by enahum**
#### Ticket Link
https://mattermost.atlassian.net/browse/RN-538

### CODE REVIEW

File: app/actions/views/file_upload.js [hunk 1]
```diff
@@ -99,7 +99,15 @@ export function retryFileUpload(file, rootId) {
     };
 }
 
-export function handleClearFiles(channelId, rootId) {
+export function handleClearFiles(channelId, rootId, onlyFailed = false) {
+    if (onlyFailed) {
```
Comment (hmhealey) [inline app/actions/views/file_upload.js]: Can you make this a separate function? I'm not a fan of ones that completely change their behaviour through a flag

Comment (enahum) [reply app/actions/views/file_upload.js]: I wanted to avoid that but sure, I can go ahead and do that now

**Comment by enahum** [comment]
@hmhealey done

**Comment by enahum** [comment]
great catch @mkraft fixed it

**Pull request merged**